hail/python/hailtop/pipeline/pipeline.py

```diff
@@ -391,7 +391,10 @@ def schedule_task(t):
                     raise PipelineException("cycle detected in dependency graph")
 
         self._tasks = ordered_tasks
-        self._backend._run(self, dry_run, verbose, delete_scratch_on_exit, **backend_kwargs)
+        try:
+            self._backend._run(self, dry_run, verbose, delete_scratch_on_exit, **backend_kwargs)
```

Comment:
Where do the retries come in? Your tests show having to call submit multiple times? Is the user calling `pipeline.run()` over again? Maybe this is in your next PR?

Reply:
Those tests are no longer present

Reply:
I still don't see how this code is resilient. Does the user have to try resubmitting again?

Reply:
I understand that /jobs/create is now idempotent. Let's say the internet is disconnected mid-way through and you get a bunch of non-transient errors. Who is calling BatchBuilder.submit() again for the jobs to be resubmitted to batch? Or is pipeline still not tolerant of random failures while submitting?

Reply:
Disconnection of an ethernet cable and disabling wifi on MacOS, in my experiments, treated as a transient error (the name `batch.hail.is` cannot be resolved). Pipeline & BatchClient are not tolerance of non-transient errors.

Reply:
Thanks! That was the part I was missing.

Reply:
> Let's say the internet is disconnected

As Dan said this is transient.  I describe this way: a transient error is one that we expect to eventually resolve itself.  Computers normally return to the internet eventually, hence it is transient.

Non-transient errors are errors that will always fail if repeated (that is, they will never resolve themselves).  For example, invalid requests.

There's probably a gray area, but for now this dichotomy seems to be working well for now.